src/pyupgrade/plugins/redundant_open_modes.rs

```diff
@@ -0,0 +1,131 @@
+use rustpython_ast::{Constant, Expr, ExprKind, Located, Location};
+use rustpython_parser::lexer;
+use rustpython_parser::token::Tok;
+
+use crate::ast::helpers::{self, match_name_or_attr};
+use crate::ast::types::Range;
+use crate::autofix::Fix;
+use crate::check_ast::Checker;
+use crate::checks::{Check, CheckCode, CheckKind};
+use crate::source_code_locator::SourceCodeLocator;
+
+const OPEN_FUNC_NAME: &str = "open";
+
+enum OpenMode {
+    U,
+    Ur,
+    Ub,
+    RUb,
+    R,
+    Rt,
+    Wt,
+}
+
+impl OpenMode {
+    fn from_str(input: &str) -> Option<OpenMode> {
+        match input {
+            "U" => Some(OpenMode::U),
+            "Ur" => Some(OpenMode::Ur),
+            "Ub" => Some(OpenMode::Ub),
+            "rUb" => Some(OpenMode::RUb),
+            "r" => Some(OpenMode::R),
+            "rt" => Some(OpenMode::Rt),
+            "wt" => Some(OpenMode::Wt),
+            _ => None,
+        }
+    }
+
+    fn replacement_value(&self) -> String {
+        match *self {
+            OpenMode::U => String::from(""),
+            OpenMode::Ur => String::from(""),
+            OpenMode::Ub => String::from("\"rb\""),
+            OpenMode::RUb => String::from("\"rb\""),
+            OpenMode::R => String::from(""),
+            OpenMode::Rt => String::from(""),
+            OpenMode::Wt => String::from("\"w\""),
+        }
+    }
+}
+
+fn match_open(expr: &Expr) -> Option<&Located<ExprKind>> {
+    if let ExprKind::Call { func, args, .. } = &expr.node {
+        if match_name_or_attr(func, OPEN_FUNC_NAME) {
```

Comment:
Can you add a TODO to validate that `open` is still bound to the built-in `open`? I.e., that it hasn't been assigned to some other function or variable? (We need to solve this in a general way, for some other rules too, so doesn't have to happen here.)

It can formatted like:

```
TODO(andberger): Verify that "open" is still bound to the built-in function.
```

(That doesn't mean you're responsible for fixing the TODO, just that you're the author and so have context on it.)

Reply:
(\cc @JonathanPlasse - another case in which we need to validate a built-in.)

Reply:
Added the TODO 👍